storage-node/src/cache/data_store_cache/memdisk/cache_manager.rs

```diff
@@ -0,0 +1,361 @@
+use futures::stream::StreamExt;
+use tokio::sync::RwLock;
+
+use crate::{
+    cache::{
+        data_store_cache::DataStoreCache,
+        policy::{DataStoreReplacer, ParpulseDataStoreCacheKey},
+    },
+    disk::disk_manager::DiskManager,
+    error::{ParpulseError, ParpulseResult},
+    storage_reader::StorageReaderStream,
+};
+
+use super::data_store::{disk::DiskStore, memory::MemStore};
+use async_trait::async_trait;
+use bytes::Bytes;
+use tokio::sync::mpsc::Receiver;
+
+/// The default maximum single file size for the memory cache.
+/// If the file size exceeds this value, the file will be stored in the disk cache.
+/// TODO(lanlou): make this value configurable.
+pub const DEFAULT_MEM_CACHE_MAX_FILE_SIZE: usize = 1024 * 512;
+
+pub struct MemDiskStoreCache<R: DataStoreReplacer> {
+    disk_store: DiskStore,
+    mem_store: Option<MemStore>,
+    /// Cache_key = S3_PATH; Cache_value = (CACHE_BASE_PATH + S3_PATH, size)
+    disk_replacer: RwLock<R>,
+    mem_replacer: Option<RwLock<R>>,
+}
```

Comment:
Maybe we can add some comments on this data structure, especially the relationship between `max_file_size` in `mem_store` and `max_capacity` in `mem_replacer`. For example, the `mem_max_file_size` in the `new` method won't come into effect in the case where `mem_max_file_size` is larger than the `max_capacity` of the `mem_replacer`.